выполнено дз 1

Ganzalo opened the pull request:
Пункт 1 выполнен
Пункт 2 выполнен
Пункт 3 выполнен

Ganzalo committed:
выполнено дз 1

Пункт 1 выполнен
Пункт 2 выполнен
Пункт 3 выполнен

usharik commented on src/ru/geekbrains/lesson1/Application.java:
```diff
@@ -13,19 +16,26 @@
+                new Dog("Бобик", Color.GRAY, 5, 100, 1, 20),
+                new Human("Джон", Color.WHITE, 20, 100, 3, 25),
+                new Robot("C-3PO", Color.YELLOW, 500, 10, 5)
+                // TODO здесь должны быть участники всех видов (Cat, Dog, Human, Robot)
```
TODO принято убирать после выполнения

usharik commented on src/ru/geekbrains/lesson1/Team.java:
```diff
@@ -1,19 +1,79 @@
+        System.out.println("Состав команды " + nameTeam);
+        System.out.println("-----------------------------------------");
+        for (Participant participant : participants) {
+            if(participant instanceof Cat) {
```
Не очень хорошая идея. Уж если хотите так выводить информацию, то нужно доработать интерфейс Participant. Фактически вы здесь нарушаете инкапсуляцию. Если у нас появляется какой-то новый тип участников то в вашем случае придется не только реализовывать этот класс, но и вносить изменения в класс Team. Интерфейс Participant как раз и сделан для того, чтобы избежать подобных лишних изменений и сделать соревновательный процесс независимым от количества типов участников которые есть или могут появиться.

Ganzalo replied:
Как вариант создать в интерфейсе Participant метод get info и реализовать для каждого класса?

usharik replied:
Да. Именно так.

Ganzalo committed:
Исправлены ошибки

- Убраны todo в коде
- Удален класс  ClassTypesDemo
- Создан метод getInfo() в Participant для вывода информации о участниках

Ganzalo committed:
1

1

Ganzalo committed:
Revert "1"

This reverts commit 5e4dd8f26b27e23d8205d512e52f1c1ea7831155.
